nusight2/src/client/components/localisation/robot_model.ts

```diff
@@ -175,4 +180,16 @@ export class LocalisationRobotModel {
   get Hft(): Matrix4 {
     return this.Hfw.multiply(this.Htw.invert());
   }
+
+  /** Field line points in field space */
+  @computed
+  get rPFf(): Vector3[] {
+    return this.fieldLinePoints.rPWw.map(rPWw => rPWw.applyMatrix4(this.Hfw));
+  }
+
+  /** Ball position in field space */
+  @computed
+  get rBFf(): Vector3 {
+    return this.ball.rBWw.applyMatrix4(this.Hfw);
+  }
```

Comment:
Which means this becomes

```suggestion
  get rBFf(): Vector3 | undefined {
    return this.ball?.rBWw.applyMatrix4(this.Hfw);
  }
```